pandas/tests/groupby/test_categorical.py

```diff
@@ -420,6 +420,18 @@ def test_observed_groups(observed):
     tm.assert_dict_equal(result, expected)
 
 
+def test_observed_groups_with_nan():
+    # GH 24740
+    df = pd.DataFrame({'cat': pd.Categorical(['a', 'c', 'a'],
```

Comment:
Could you make this `pd.Categorical(['a', np.nan, 'a'], categories=['a', 'b, 'd', 'e', 'f'])`? It wasn't immediately obvious that ''c'` would end up as NA.